clients/src/main/java/org/apache/kafka/common/serialization/Deserializer.java

```diff
@@ -32,13 +32,13 @@
     public void configure(Map<String, ?> configs, boolean isKey);
     
     /**
-     *
+     * Deserialize a record value from a bytearray into a value or object.
      * @param topic topic associated with the data
-     * @param data serialized bytes
-     * @return deserialized typed data
+     * @param data serialized bytes; may be null. If null, implementations should return null.
```

Comment:
Implementations are free to deserialize a null into anything that makes sense to them, right?